src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int randomValue = 100;
+    private static final String defaultColor = "WHITE";
+    private static final int defaultRadius = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure generateFigure() {
+        switch (FigureType.values()[random.nextInt(FigureType.values().length)]) {
+            case CIRCLE:
+                return getRandomCircle();
+            case SQUARE:
+                return getRandomSquare();
+            case RIGHT_TRIANGLE:
+                return getRandomRightTriangle();
+            case ISOSCELES_TRAPEZOID:
+                return getRandomIsoscelesTrapezoid();
+            case RECTANGLE:
+                return getRandomRectangle();
+            default:
+                throw new RuntimeException("Figure does not exist");
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(defaultColor,defaultRadius);
+    }
+
+    private int getRandomNumber() {
+        return random.nextInt(randomValue);
+    }
+
+    public Figure getRandomSquare() {
+        return new Square(colorSupplier.getRandomColor(),getRandomNumber());
+    }
```

Comment:
Other 'getRandom...' methods should also be private because they are not used outside this class
```suggestion
    private Figure getRandomSquare() {
        return new Square(colorSupplier.getRandomColor(), getRandomNumber());
    }
```